services/lumberjack/src/service.js

```diff
@@ -0,0 +1,255 @@
+    this._telemetryTask =
+      createTimeoutTask(this._telemetryOverview.bind(this),
+        this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+  }
+
+  /** Get service ping data and write to the database */
+  async _pingTask() {
+    //Get ping data
+    let ping =
+      (await request.get('http://' + this._pingHost + ':' +
+        this._pingPort + '/api/ping')
+        .proto(stats.PingTimes)
+        .timeout(1000)).body;
+
+    //Write data for services
+    for (let endpoint of ping.service_pings) {
+      let { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: 'lumberjack'
+      });
+    }
+
+    //Write data for devices
+    for (let endpoint of ping.device_pings) {
+      let { host, port, name} = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { devicePing: endpoint.ms },
+          tags: { host, port, name}
+        }], {
+        database: 'lumberjack'
+      });
+    }
+  }
+
+  /** Get telemetry upload rate data and write to the database */
+  async _uploadRate() {
+    //Get upload rate
+    let rate =
+      (await request.get('http://' + this._forwardInteropHost + ':' +
+        this._forwardInteropPort + '/api/upload-rate')
+        .proto(stats.InteropUploadRate)
+        .timeout(1000)).body;
+
+    await this._influx.writeMeasurement('upload-rate', [
+      {
+        fields: { t1: rate.total_1, t5: rate.total_5,
+          f1: rate.fresh_1, f5: rate.fresh_5 },
+        tags: { host: this._forwardInteropHost, port: this._forwardInteropPort }
+      }]);
+  }
+
+  /** Get telemetry overview and task queue length and
+  write to the database */
+  async _telemetryOverview() {
+    const OFFLINE = 0;
+    const ONLINE = 1;
+    let gstatus, pstatus;
+
+    //Get telemetry overview
+    let groundTelem =
+      (await request.get('http://' + this._telemetryHost + ':' +
+        this._telemetryPort + '/api/overview')
+        .proto(telemetry.Overview)
+        .timeout(1000)).body;
+
+    //Get length of task queue for the plane
+    let queueLength =
+      (await request.get('http://' + this._telemetryHost + ':' +
+        this._telemetryPort + '/api/queue-length')
+        .timeout(1000)).body;
+
+    //Check if current time is the same or less than the previous
+    //timestate
+    if (groundTelem.time <= gTimes) {
+      gstatus = OFFLINE;
+      pstatus = OFFLINE;
+    } else {
+      gstatus = ONLINE;
+      pstatus = ONLINE;
+    }
+    if (queueLength > this._queueLimit){
+      pstatus = OFFLINE;
+    }
+
+    //update current time if greater or same from previous time state
+    if (groundTelem.time >= gTimes)
+      gTimes = groundTelem.time;
+
+    await this._influx.writeMeasurement('telemetry', [
+      {
+        fields: { gstatus: gstatus, pstatus: pstatus },
```

Comment:
you can collapse these like so:
```suggestion
        fields: { gstatus, pstatus },
```